### PR TITLE
Prepare new release: v0.18.1

### PR DESCRIPTION
**Pull request by behackl**
Release preparations for v0.18.1.

For this version, we will not generate a separate changelog page in our documenation, and instead just use the autogenerated changelog from the release page.

(We still need to sit down and edit labels + titles of merged PRs to make the autogenerated notes somewhat readable.)

---

### Breaking Changes and Deprecations

* Removed deprecated `manim new` command by @chopan050 in https://github.com/ManimCommunity/manim/pull/3512
* Removed support for dynamic plugin imports by @Viicos in https://github.com/ManimCommunity/manim/pull/3524
* Remove meth:``.Mobject.wag`` by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3539
* Remove deprecated parameters and animations by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3688

### New Features

* Added `cap_style` feature to `VMobject` by @MathItYT in https://github.com/ManimCommunity/manim/pull/3516
* Allow hiding version splash by @jeertmans in https://github.com/ManimCommunity/manim/pull/3329
* Added the ability to pass lists and generators to `Scene.play()` by @MrDiver in https://github.com/ManimCommunity/manim/pull/3365
* Added ``--preview_command`` cli flag by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3615

### Fixed Bugs and Enhancements

* Allow accessing ghost vectors in :class:`.LinearTransformationScene` by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3435
* Optimized `get_unit_normal()` and replaced `np.cross()` with custom `cross()` in `manim.utils.space_ops` by @chopan050 in https://github.com/ManimCommunity/manim/pull/3494
* Implement caching of fonts list to improve runtime performance by @MrDiver in https://github.com/ManimCommunity/manim/pull/3316
* Reformatting the `--save_sections` output to have the format `<Scene>_<SecNum>_<SecName><extension>` by @doaamuham in https://github.com/ManimCommunity/manim/pull/3499
* Account for dtype in the pixel array so the maximum value stays correct in the invert function by @jeertmans in https://github.com/ManimCommunity/manim/pull/3493
* Added `grid_lines` attribute to `Rectangle` to add individual styling to the grid lines by @RobinPH in https://github.com/ManimCommunity/manim/pull/3428
* Fixed rectangle grid properties (#3082) by @pauluhlenbruck in https://github.com/ManimCommunity/manim/pull/3513
* Fixed animations with zero runtime length to give a useful error instead of a broken pipe by @MrDiver in https://github.com/ManimCommunity/manim/pull/3491
* Fixed stroke width being ignored by `StreamLines` with a single color by @yashm277 in https://github.com/ManimCommunity/manim/pull/3436
* Fixed formatting of ``MoveAlongPath`` docs by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3541
* Added helpful hints to `VGroup.add()` error message by @vvolhejn in https://github.com/ManimCommunity/manim/pull/3561
* Made `earclip_triangulation` more robust by @hydromelvictor in https://github.com/ManimCommunity/manim/pull/3574
* Refactored `TexTemplate` by @Viicos in https://github.com/ManimCommunity/manim/pull/3520
* Fixed `write_subcaption_file` error when using OpenGL renderer by @yuan-xy in https://github.com/ManimCommunity/manim/pull/3546
* Fixed `get_arc_center()` returning reference of point by @sparshg in https://github.com/ManimCommunity/manim/pull/3599
* Improved handling of specified font name by @staghado in https://github.com/ManimCommunity/manim/pull/3429
* Fixing the behavior of `.become` to not modify target mobject via side effects fix color linking by @MrDiver in https://github.com/ManimCommunity/manim/pull/3508
* Fixed bug in :class:`.VMobjectFromSVGPath` by @abul4fia in https://github.com/ManimCommunity/manim/pull/3677
* Fix for windows cp1252 encoding failure (fix test pipeline) by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3687
* Fix NameError in try... except by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3694
* Fix successive calls of :meth:`.LinearTransformationScene.apply_matrix` by @SirJamesClarkMaxwell in https://github.com/ManimCommunity/manim/pull/3675
* Fixed `Mobject.put_start_and_end_on` with same start and end point by @MontroyJosh in https://github.com/ManimCommunity/manim/pull/3718
* Fixed issue where `SpiralIn` doesn't show elements by @Gixtox in https://github.com/ManimCommunity/manim/pull/3589
* Cleaned `Graph` layouts and increase flexibility by @Nikhil-42 in https://github.com/ManimCommunity/manim/pull/3434
* `AnimationGroup`: optimized `interpolate()` and fixed alpha bug on `finish()` by @chopan050 in https://github.com/ManimCommunity/manim/pull/3542
* Fixed warning about missing plugin `""` by @behackl in https://github.com/ManimCommunity/manim/pull/3734

### Documentation

* Typo in `indication` documentation by @jcep in https://github.com/ManimCommunity/manim/pull/3477
* Fixed typo: 360° to 180° in quickstart tutorial by @szchixy in https://github.com/ManimCommunity/manim/pull/3498
* Fixed typo in mobject docstring: `line` -> `square` by @yuan-xy in https://github.com/ManimCommunity/manim/pull/3509
* Explain ``.Transform`` vs ``.ReplacementTransform`` in quickstart examples by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3500
* Fixed formatting in building blocks tutorial by @MrDiver in https://github.com/ManimCommunity/manim/pull/3515
* Fixed `Indicate` docstring typo by @Lawqup in https://github.com/ManimCommunity/manim/pull/3461
* Added Documentation to `.to_edge` and `to_corner` by @TheMathematicFanatic in https://github.com/ManimCommunity/manim/pull/3408
* Added some words about Cairo 1.18 by @jeertmans in https://github.com/ManimCommunity/manim/pull/3530
* Fixed typo of `get_y_axis_label` parameter documentation by @yuan-xy in https://github.com/ManimCommunity/manim/pull/3547
* Added note in docstring of `ManimColor` about class constructors by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3554
* Improve documentation section about contributing to docs by @chopan050 in https://github.com/ManimCommunity/manim/pull/3555
* Removed duplicated documentation for -s / --save_last_frame CLI flag by @Gixtox in https://github.com/ManimCommunity/manim/pull/3528
* Updated Docker instructions to use bash from the PATH by @NotWearingPants in https://github.com/ManimCommunity/manim/pull/3582
* Fixed typo in `value_tracker.py` by @yuan-xy in https://github.com/ManimCommunity/manim/pull/3594
* Added `ref_class` for `BooleanOperations` in Example Gallery by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3598
* Changed `Vector3` to `Vector3D` in contributing docs by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3639
* Added some examples for `Mobject`/`VMobject` methods by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3641
* Fixed broken link to Poetry's installation guide in the documentation by @biinnnggggg in https://github.com/ManimCommunity/manim/pull/3692
* Fixed minor grammatical errors found in the index page of the documentation by @biinnnggggg in https://github.com/ManimCommunity/manim/pull/3690
* Fixed typo on page about translations by @biinnnggggg in https://github.com/ManimCommunity/manim/pull/3696
* Fixed outdated description of CLI option in Manim's Output Settings by @HairlessVillager in https://github.com/ManimCommunity/manim/pull/3674
* Mention pixi in installation guide by @pavelzw in https://github.com/ManimCommunity/manim/pull/3678
* Updated typing guidelines by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3704
* Updated documentation and typings for `ParametricFunction` by @danielzsh in https://github.com/ManimCommunity/manim/pull/3703
* Fixed docstring markup in `Rotate` by @TheCrowned in https://github.com/ManimCommunity/manim/pull/3721
* Improve consistency in axis label example by @amrear in https://github.com/ManimCommunity/manim/pull/3730

### Maintenance and Testing

* Fixed wrong path in action building downloadable docs by @behackl in https://github.com/ManimCommunity/manim/pull/3450
* Add type hints to `_config` by @Viicos in https://github.com/ManimCommunity/manim/pull/3440
* Update dependency constraints, fix deprecation warnings by @Viicos in https://github.com/ManimCommunity/manim/pull/3376
* Update Docker base image to python3.12-slim (#3458) by @PikaBlue107 in https://github.com/ManimCommunity/manim/pull/3459
* Fixed `line_join` to `joint_type` in example_scenes/basic.py by @szchixy in https://github.com/ManimCommunity/manim/pull/3510
* Fixed :attr:`.Mobject.animate` type-hint to allow LSP autocomplete by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3543
* Finish TODO's in ``contributing/typings.rst`` by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3545
* Fixed use of `Mobject`'s deprecated `get_*()` and `set_*()` methods in Cairo tests by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3549
* Added support for Manim type aliases in Sphinx docs and added new TypeAliases by @chopan050 in https://github.com/ManimCommunity/manim/pull/3484
* Fixed typing of `Animation` by @dandavison in https://github.com/ManimCommunity/manim/pull/3568
* Added some TODOs for future use of `ManimFrame` by @chopan050 in https://github.com/ManimCommunity/manim/pull/3553
* Fixed typehint of :attr:`InternalPoint2D_Array` by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3592
* Fixed error in Windows CI pipeline by @behackl in https://github.com/ManimCommunity/manim/pull/3611
* Fixed type hint of indication.py by @yuan-xy in https://github.com/ManimCommunity/manim/pull/3613
* Revert vector type aliases to NumPy ndarrays by @chopan050 in https://github.com/ManimCommunity/manim/pull/3595
* Run `poetry lock --no-update` by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3621
* Code Cleanup: removing unused imports and global variables by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3620
* Fixed type hint of `Vector` direction parameter by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3640
* Flake8 rule C901 is about McCabe code complexity by @cclauss in https://github.com/ManimCommunity/manim/pull/3673
* Updated year in license by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3689
* Automated copyright updating for docs by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3708
* Fixed some typehints in `mobject.py` by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3668
* Search for type aliases if TYPE_CHECKING by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3671
* Follow-up to graph layout cleanup: improvements for tests and typing by @behackl in https://github.com/ManimCommunity/manim/pull/3728
* GH Actions: Changed from macos-latest to macos-13 by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3729
* Fixed return type inconsistency for `get_anchors()` by @JinchuLi2002 in https://github.com/ManimCommunity/manim/pull/3214
* Prepared new release: `v0.18.1` by @behackl in https://github.com/ManimCommunity/manim/pull/3719

#### Dependency Version Changes

* Bump jupyter-server from 2.9.1 to 2.11.2 by @dependabot in https://github.com/ManimCommunity/manim/pull/3497
* Bump github/codeql-action from 2 to 3 by @dependabot in https://github.com/ManimCommunity/manim/pull/3567
* Bump actions/upload-artifact from 3 to 4 by @dependabot in https://github.com/ManimCommunity/manim/pull/3566
* Bump actions/setup-python from 4 to 5 by @dependabot in https://github.com/ManimCommunity/manim/pull/3565
* updated several packages (pillow, jupyterlab, notebook, jupyterlab-lsp, jinja2, gitpython) by @behackl in https://github.com/ManimCommunity/manim/pull/3593
* Update jupyter.rst by @abul4fia in https://github.com/ManimCommunity/manim/pull/3630
* Bump black from 23.12.1 to 24.3.0 by @dependabot in https://github.com/ManimCommunity/manim/pull/3649
* Bump cryptography from 42.0.0 to 42.0.4 by @dependabot in https://github.com/ManimCommunity/manim/pull/3629
* Bump actions/cache from 3 to 4 by @dependabot in https://github.com/ManimCommunity/manim/pull/3607
* Bump FedericoCarboni/setup-ffmpeg from 2 to 3 by @dependabot in https://github.com/ManimCommunity/manim/pull/3608
* Bump ssciwr/setup-mesa-dist-win from 1 to 2 by @dependabot in https://github.com/ManimCommunity/manim/pull/3609
* Bump idna from 3.6 to 3.7 by @dependabot in https://github.com/ManimCommunity/manim/pull/3693
* Bump pillow from 10.2.0 to 10.3.0 by @dependabot in https://github.com/ManimCommunity/manim/pull/3672
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/ManimCommunity/manim/pull/3332
* Updated sphinx deps by @JasonGrace2282 in https://github.com/ManimCommunity/manim/pull/3720

## New Contributors
* @Lawqup made their first contribution in https://github.com/ManimCommunity/manim/pull/3461
* @jcep made their first contribution in https://github.com/ManimCommunity/manim/pull/3477
* @szchixy made their first contribution in https://github.com/ManimCommunity/manim/pull/3498
* @PikaBlue107 made their first contribution in https://github.com/ManimCommunity/manim/pull/3459
* @yuan-xy made their first contribution in https://github.com/ManimCommunity/manim/pull/3509
* @MathItYT made their first contribution in https://github.com/ManimCommunity/manim/pull/3516
* @doaamuham made their first contribution in https://github.com/ManimCommunity/manim/pull/3499
* @RobinPH made their first contribution in https://github.com/ManimCommunity/manim/pull/3428
* @pauluhlenbruck made their first contribution in https://github.com/ManimCommunity/manim/pull/3513
* @yashm277 made their first contribution in https://github.com/ManimCommunity/manim/pull/3436
* @TheMathematicFanatic made their first contribution in https://github.com/ManimCommunity/manim/pull/3408
* @vvolhejn made their first contribution in https://github.com/ManimCommunity/manim/pull/3561
* @hydromelvictor made their first contribution in https://github.com/ManimCommunity/manim/pull/3574
* @dandavison made their first contribution in https://github.com/ManimCommunity/manim/pull/3568
* @Gixtox made their first contribution in https://github.com/ManimCommunity/manim/pull/3528
* @staghado made their first contribution in https://github.com/ManimCommunity/manim/pull/3429
* @biinnnggggg made their first contribution in https://github.com/ManimCommunity/manim/pull/3692
* @HairlessVillager made their first contribution in https://github.com/ManimCommunity/manim/pull/3674
* @SirJamesClarkMaxwell made their first contribution in https://github.com/ManimCommunity/manim/pull/3675
* @danielzsh made their first contribution in https://github.com/ManimCommunity/manim/pull/3703
* @TheCrowned made their first contribution in https://github.com/ManimCommunity/manim/pull/3721
* @MontroyJosh made their first contribution in https://github.com/ManimCommunity/manim/pull/3718
* @amrear made their first contribution in https://github.com/ManimCommunity/manim/pull/3730

**Full Changelog**: https://github.com/ManimCommunity/manim/compare/v0.18.0.post0...v0.18.1

### CODE REVIEW

**Review by JasonGrace2282 — APPROVED** [review]
LGTM! I'm excited about the release :)

**Comment by naveen521kk** [comment]
https://github.com/ManimCommunity/manim/pull/3524 - should be a breaking change imo

**Comment by behackl** [comment]
> #3524 - should be a breaking change imo

Agreed -- changed!

**Comment by jeertmans** [comment]
Impressive!

If there are breaking changes, shouldn’t the version be 0.19?

**Comment by behackl** [comment]
> Impressive!
> 
> If there are breaking changes, shouldn’t the version be 0.19?

Thanks! Technically, this just removes a bunch of deprecated code, plus the removal of the "dynamic plugins" -- which I do not think is a particularly serious breaking change, so it's fine to keep this as v0.18.1.

(v0.19.0 isn't too far away though, looking at #3501 ... :-))

File: CITATION.cff [hunk 1]
```diff
@@ -4,10 +4,10 @@ authors:
   -
     name: "The Manim Community Developers"
 cff-version: "1.2.0"
-date-released: 2023-11-11
+date-released: 2024-04-23
```
Comment (behackl) [inline CITATION.cff:7]:
```suggestion
date-released: 2024-04-28
```

**Comment by jeertmans** [comment]
> > Impressive!
> > If there are breaking changes, shouldn’t the version be 0.19?
> 
> Thanks! Technically, this just removes a bunch of deprecated code, plus the removal of the "dynamic plugins" -- which I do not think is a particularly serious breaking change, so it's fine to keep this as v0.18.1.
> 
> (v0.19.0 isn't too far away though, looking at #3501 ... :-))

OOh ok that's why I didn't see #3501 in the release notes, because this is for next release!

**Comment by michol92** [comment]
After updating to v0.18.1, there is an error like this
Is there any solution?
macbook pro

Traceback (most recent call last):
  File "/Users/michol/opt/anaconda3/bin/manim", line 5, in <module>
    from manim.__main__ import main
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/__init__.py", line 104, in <module>
    from .utils.ipython_magic import ManimMagic
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/utils/ipython_magic.py", line 13, in <module>
    from manim.__main__ import main
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/__main__.py", line 11, in <module>
    from .cli.plugins.commands import plugins
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/cli/plugins/commands.py", line 14, in <module>
    from ...plugins.plugins_flags import list_plugins
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/plugins/__init__.py", line 13, in <module>
    missing_plugins = requested_plugins - set(get_plugins().keys())
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/plugins/plugins_flags.py", line 19, in get_plugins
    plugins: dict[str, Any] = {
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim/plugins/plugins_flags.py", line 20, in <dictcomp>
    entry_point.name: entry_point.load()
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/importlib_metadata/__init__.py", line 203, in load
    module = import_module(match.group('module'))
  File "/Users/michol/opt/anaconda3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim_data_structures/__init__.py", line 3, in <module>
    from .m_array import *
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim_data_structures/m_array.py", line 527, in <module>
    class MArray(VGroup):
  File "/Users/michol/opt/anaconda3/lib/python3.9/site-packages/manim_data_structures/m_array.py", line 630, in MArray
    def __calc_label_pos_and_mob(self) -> typing.Tuple[Square, np.ndarray]:
AttributeError: module 'manim.typing' has no attribute 'Tuple'

**Comment by jeertmans** [comment]
Ho yeah Python 3.13 remove a few things from the typing module.

But I don’t think Python 3.13 is officially supported by Manim, is it?

**Pull request merged**